mars-sim-core/src/main/java/com/mars_sim/core/environment/TerrainElevation.java

```diff
@@ -287,233 +263,6 @@ public double obtainRegolithCollectionRate(Coordinates loc) {
-		
-        // Patch elevation problems at certain locations.
-//		height = patchElevation(height, phi, theta);
-		
-		return height;
-	}
-	
-    /**
-     * Patches elevation errors around mountain tops.
-     * @param elevation the original elevation for the location.
-     * @param location the coordinates
-     * @return the patched elevation for the location
-     */
-    private static double patchElevation(double elevation, double phi, double theta) {
-    	double result = elevation;
-		// Patch errors at Olympus Mons caldera.
-    	
-		// Patch the smallest cauldera at the center
-		if (Math.abs(theta - OLYMPUS_MONS_CALDERA_THETA) < .0176
-			 && Math.abs(phi - OLYMPUS_MONS_CALDERA_PHI) < .0174) {
-				result = 19; 
-		}
-		
-		// Patch the larger white cauldera 
-		else if (Math.abs(theta - OLYMPUS_MONS_CALDERA_THETA) < .0796
-			&& Math.abs(phi - OLYMPUS_MONS_CALDERA_PHI) < .0796) {
-
-			if (elevation > 19 && elevation < 21.2870)
-				result = elevation;
-			else
-				result = 21.287D;
-		}
-		
-		// Patch the red base cauldera 
-		else if (Math.abs(theta - OLYMPUS_MONS_CALDERA_THETA) < .1731
-			&& Math.abs(phi - OLYMPUS_MONS_CALDERA_PHI) < .1731) {
-
-			if (elevation < 19 && elevation > 3)
-				result = elevation;
-			else
-				result = 3;
-		}
-		
-    	// Patch errors at Ascraeus Mons.
-		if (Math.abs(theta - ASCRAEUS_MONS_THETA) < .02D) {
-			if (Math.abs(phi - ASCRAEUS_MONS_PHI) < .02D) {
-				if (elevation < 3D) result = 20D;
-			}
-		}
-    	
-		else if (Math.abs(theta - ARSIA_MONS_THETA) < .04D) {
-			if (Math.abs(phi - ARSIA_MONS_PHI) < .04D) {
-				if (elevation < 3D)
-					result = 17.781;
-			}
-		}
-		
-		else if (Math.abs(theta - ELYSIUM_MONS_THETA) < .04D) {
-			if (Math.abs(phi - ELYSIUM_MONS_PHI) < .04D) {
-				if (elevation < 3D)
-					result = 14.127;
-			}
-		}
-		
-		else if (Math.abs(theta - PAVONIS_MONS_THETA) < .04D) {
-			if (Math.abs(phi - PAVONIS_MONS_PHI) < .04D) {
-				if (elevation < 3D)
-					result = 14.057;
-			}
-		}
-		
-		else if (Math.abs(theta - HECATES_THOLUS_THETA) < .04D) {
-			if (Math.abs(phi - HECATES_THOLUS_PHI) < .04D) {
-//				if (elevation < 2.5D)
-					result = 4.853;
-			}
-		}
-		
-		// Patch errors at Ascraeus Mons.
-		else if (Math.abs(theta - ALBOR_THOLUS_THETA) < .04D) {
-			if (Math.abs(phi - ALBOR_THOLUS_PHI) < .04D) {
-//				if (elevation < 2D)
-					result = 3.925;
-			}
-		}
-		
-//		// Patch errors at the north pole.
-//		else if (Math.abs(location.getTheta() - NORTH_POLE_THETA) < .2D) {
-//			if (Math.abs(location.getPhi() - NORTH_POLE_PHI) < .04D) {
-////				if (elevation < 2D)
-//					result = 1.015;
-//			}
-//		}
-//		
-//		// Patch errors at the south pole.
-//		else if (Math.abs(location.getTheta() - SOUTH_POLE_THETA) < .04D) {
-//			if (Math.abs(location.getPhi() - SOUTH_POLE_PHI) < .04D) {
-////				if (elevation < 2D)
-//					result = .783;
-//			}
-//		}
-          
-    	return result;
-    }
-    
-	public int getHue(int red, int green, int blue) {

```

Comment:
Note that the Hue (H) from HSB value encodes the height information under the DEM/DTM model. 

That's why we are experimenting this approach here as explained in this [thread](https://github.com/mars-sim/mars-sim/issues/916).